### ISSUE TITLE
Metadata Cache Support For Presto Iceberg Connector

### ISSUE THREAD

**Issue by agrawalreetika** [prestodb/presto#20726]
Currently, In Presto Iceberg connector metadata is not being cached so every time it will fetch the metadata-related information from the File System for the Iceberg catalog.

It would be good to add support for Metadata cache in the Iceberg Connector.

**Comment by ZacBlanco** [comment]
On a related note, table statistics are also not cached. Since table versions are immutable it should be straightforward to implement a cache for that as well.

**Comment by agrawalreetika** [comment]
Yes, I had also put up some details for Caching stats w.r.t Iceberg Connector here - https://github.com/prestodb/presto/issues/20725
I think Caching stats could be a generic implementation for any Lakehouse table connector though.

**Comment by tdcmeehan** [comment]
Manifest file caching can be quickly picked up in Presto by leveraging the work here: https://github.com/apache/iceberg/pull/4518

It appears that Catalog caching doesn't quickly invalidate, and I am trying to understand why this is.

**Comment by nmahadevuni** [comment]
> Manifest file caching can be quickly picked up in Presto by leveraging the work here: [apache/iceberg#4518](https://github.com/apache/iceberg/pull/4518)
> 
> It appears that Catalog caching doesn't quickly invalidate, and I am trying to understand why this is.

Created https://github.com/prestodb/presto/issues/20820 to track manifest file caching.

**Comment by fuyun2024** [comment]
I noticed that Iceberg has already supported this feature, I wonder if Delta Lake can also support it?

**Comment by agrawalreetika** [comment]
@fuyun2024,
Try setting the below properties in your Delta catalog to enable metadata caching in the Delta connector:
```
hive.metastore-cache-ttl=
hive.metastore-refresh-interval=
```

**Comment by agrawalreetika** [comment]
@steveburnett, Actually metadata cache is not fully supported in the Delta & Iceberg connector because of the reason I mentioned above it's supported. So it works in a very limited way for these connectors, so I am not sure if we want to add these as of now in Delta doc. Metadata cache is completely supported for Hive connector so it's mentioned there but is limited to other lakehosue tables.

**Comment by steveburnett** [comment]
Thanks @agrawalreetika, that's everything I wanted to know! Makes complete sense, I was just wondering.

**Comment by fuyun2024** [comment]
> @fuyun2024 Try setting the below properties in your Delta catalog to enable metadata caching in the Delta connector:
> 
> ```
> hive.metastore-cache-ttl=
> hive.metastore-refresh-interval=
> ```
> 
> **Note:** But I think for Iceberg (Hive Catalog) & Delta connector, this would only work for Schema and Table Names since that's the information being fetched from Metastore. And other metadata would be fetched from the filesystem Metastore Cache which is present for hive may not take effect for those.

Thanks @agrawalreetika,  Following your advice, I've made some adjustments and indeed, the query speed has improved, with planning time reduced from 5-7 seconds to within 2 seconds. However, as you've mentioned, this would only work for Schema and Table Names. I wonder if we are going to support caching the delta meta  file list (including the maximum and minimum values of the file)?

**Comment by tdcmeehan** [comment]
@agrawalreetika does it make sense to at least document the limited support with Hive Metastore that exists today?  Clearly there is a positive impact to query analysis timing.

**Comment by agrawalreetika** [comment]
@tdcmeehan I was thinking, we could document all the available RaptorX caching that each connector (Delta, Iceberg & Hive) supports along with their limitations on the respective connector page. 
What do you think?

**Comment by agrawalreetika** [comment]
Found the issue created - https://github.com/prestodb/presto/issues/21512 👍🏻

**Comment by tdcmeehan** [comment]
> @tdcmeehan I was thinking, we could document all the available RaptorX caching that each connector (Delta, Iceberg & Hive) supports along with their limitations on the respective connector page. What do you think?

Sounds good to me!
